### DISCUSSION TITLE
dbgenerated - how to set default if another column has a value? (Postgres)

### DISCUSSION THREAD

**Discussion by Extarys** [prisma/prisma#19690] [Q&A] [answered]
I never thought about using `dbgenerated` before and I'm unsure how to use it or return a value.
I have:
```
	time  Int?
	totalTime Int?
```
I need `time` to default to `0` if `timeTotal` is not null. As a default value for `time` I guess the condition might look something like that:
`@default(dbgenerated("(totalTime <> NULL) IS TRUE"))`, but I don't know after that how to return an actual value :laughing: 

I know it might look simple, but I really don't know. Thanks! :D

**Comment by nurul3101** [comment, marked as answer]
Hello @Extarys 👋 

It looks like you want to set a default value for the `time` field based on the value of another field in the same row `totalTime`.
Based on my understanding, default values in SQL are typically static and can't depend on the value of other fields in the same row.

One of the straightforward way would be to handle this logic at application level.
You could do something like this:

```typescript
async function createTimeRecord(data: Partial<TimeRecord>) {
  const time = data.totalTime != null ? 0 : data.time;
  return prisma.timeRecord.create({
    data: {
      ...data,
      time,
    },
  });
}
```

In this snippet, we check that if `totalTime` is not null, time will be set to 0, otherwise, it will be set to the value provided in the data. 

Hope this helps! Please feel free to let us know in case you have any additional queries. If not, please mark this discussion as Answered so that we are aware that this is resolved.

**Reply by Extarys** [reply]
Thanks! I'll indeed need to do it app side. It's not a big problem, but I'm trying to find ways to avoid mistakes on the app side (forgetting to decrement or set a value somewhere)

**Reply by nurul3101** [reply]
You can write tests to ensure that the conditions works as expected 😄
